programs/ui-wrapper/src/processors/place_order.rs

```diff
@@ -0,0 +1,260 @@
+use std::{
+    cell::{Ref, RefMut},
+    mem::size_of,
+};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use hypertree::{
+    get_helper, get_mut_helper, trace, DataIndex, FreeList, HyperTreeReadOperations,
+    HyperTreeWriteOperations, RBNode, NIL,
+};
+use manifest::{
+    program::{
+        batch_update::{BatchUpdateReturn, PlaceOrderParams}, batch_update_instruction, deposit_instruction, expand_market_instruction, get_mut_dynamic_account
+    },
+    quantities::{BaseAtoms, QuoteAtoms, QuoteAtomsPerBaseAtom, WrapperU64},
+    state::{DynamicAccount, MarketFixed, OrderType, NO_EXPIRATION_LAST_VALID_SLOT},
+    validation::{ManifestAccountInfo, Program, Signer},
+};
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    program::{get_return_data, invoke},
+    pubkey::Pubkey,
+    system_program,
+};
+
+use crate::{
+    market_info::MarketInfo, open_order::WrapperOpenOrder, wrapper_state::ManifestWrapperStateFixed,
+};
+
+use super::shared::{
+    check_signer, expand_wrapper_if_needed, get_market_info_index_for_market, sync_fast,
+    OpenOrdersTree, UnusedWrapperFreeListPadding, WrapperStateAccountInfo,
+};
+
+#[derive(BorshDeserialize, BorshSerialize, Clone)]
+pub struct WrapperPlaceOrderParams {
+    client_order_id: u64,
+    base_atoms: u64,
+    price_mantissa: u32,
+    price_exponent: i8,
+    is_bid: bool,
+    last_valid_slot: u32,
+    order_type: OrderType,
```

Comment:
should ordertype be different to allow in the future adding FillOrKill and PostOnlySlide

Reply:
its u8, very extensible